src/binary-search-tree.js

```diff
@@ -32,8 +74,26 @@ class BinarySearchTree {
   // again. Whatever floats your boat.
   /* eslint-disable global-require */
   breadthFirstForEach(cb) {
```

Comment:
Your implementation seems a bit convoluted. Feel free to refer to the solution repository to help you figure out what you need and don't need in your implementation of this method.